src/psyclone/tests/psyir/backend/sir_test.py

```diff
@@ -942,4 +940,83 @@ def test_sirwriter_schedule_node_1(parser, sir_writer):
         "  make_literal_access_expr(\"1.0\", BuiltinType.Float),\n"
         "  \"=\")," in schedule_result)
 
+
+def test_sirwriter_unaryoperation_intrinsic_node(parser, sir_writer):
+    '''Check the unaryoperation_node method of the SIRWriter class
+    outputs the expected SIR code when the unaryoperation is an
+    intrinsic.
+
+    '''
+    code = CODE.replace("1.0", "abs(1.0)")
+    rhs = get_rhs(parser, code)
+    result = sir_writer.unaryoperation_node(rhs)
+    assert ("make_fun_call_expr(\"math::fabs\", [make_literal_access_expr("
+            "\"1.0\", BuiltinType.Float)])" in result)
+
+
+@pytest.mark.parametrize("operation", ["min", "max"])
+def test_sirwriter_binaryoperation_intrinsic_node(parser, sir_writer,
+                                                  operation):
+    '''Check the binaryoperation_node method of the SIRWriter class
+    outputs the expected SIR code when the binaryoperation is an
+    intrinsic.
+
+    '''
+    code = CODE.replace("1.0", f"{operation}(1.0, 2.0)")
+    rhs = get_rhs(parser, code)
+    result = sir_writer.binaryoperation_node(rhs)
+    assert (f"make_fun_call_expr(\"math::{operation}\", ["
+            f"make_literal_access_expr(\"1.0\", BuiltinType.Float)], "
+            f"[make_literal_access_expr(\"2.0\", BuiltinType.Float)])"
+            in result)
+
+
+def test_sirwriter_binaryoperation_sign_node(parser, sir_writer):
+    '''Check the binaryoperation_node method of the SIRWriter class
+    outputs the expected SIR code when the binaryoperation is sign
+    intrinsic. This is a special case as the sign intrinsic is
+    implemented differently in Fortran and C.
```

Comment:
Picky but really we're only talking about PSyIR and SIR here?

Reply:
Modified.